### ISSUE TITLE
memberLike modifiers

### ISSUE THREAD

**Issue by pmiatkowski** [typescript-eslint/typescript-eslint#3753]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/tslint/config": ["warn", {
      "rules": {
        '@typescript-eslint/naming-convention': [
            'warn',
            { selector: 'default', format: ['camelCase'] },
            { selector: 'memberLike', modifiers: ['private', 'protected'], format: ['camelCase'], leadingUnderscore: 'require' },
        ],
      },
    }],
  }
}
```

```TS
// your repro code case
```

**Expected Result**

Eslint should not report private and protected as invalid when there is single rule defined with two modifiers:
modifiers: ['private', 'protected']

**Actual Result**

Example class:
```
export class ExampleClass {
    public prop: number;

    private _privProp: string = 'abc';

    protected _protectedProp: string = 'xyz';

    protected _methodName(): void {

    }
}
```

I get error/warning from eslint when validating:
```
Property name `_privProp` must match one of the following formats: camelCase eslint@typescript-eslint/naming-convention
``` 

All properties in the class marked as private and protected are reported by eslint.

But when I add two rules with separate modifiers, it works fine. I need to divide rules per modifiers

```
{ selector: 'memberLike', modifiers: ['protected'], format: ['camelCase'], leadingUnderscore: 'require' },
 { selector: 'memberLike', modifiers: ['private'], format: ['camelCase'], leadingUnderscore: 'require' },
```

**Additional Info**

Did eslint throw an exception? No.

**Versions**

    "@typescript-eslint/eslint-plugin": "^3.10.1",
    "@typescript-eslint/parser": "^3.10.1",

**Comment by bradzacher** [comment]
> I get error/warning.

It's generally helpful to provide the message.
How can I help you if you're only telling me a fraction of the story?
Help me to help you.

I just tested your "bad" config against master and it's working completely fine. So again - provide me some information please so I can help.

**Comment by pmiatkowski** [comment]
Sorry, 
So I want to force underscore for protected and private methods of a class.
Thing is I can't do this in a single rule using array of modfiers ['private', 'protected']
`{ selector: 'memberLike', modifiers: ['private', 'protected'], format: ['camelCase'], leadingUnderscore: 'require' },`

The rule was omitted and I was getting an error. But when I added same rule but with separated modifiers, it worked:
`{ selector: 'memberLike', modifiers: ['protected'], format: ['camelCase'], leadingUnderscore: 'require' },
{ selector: 'memberLike', modifiers: ['private'], format: ['camelCase'], leadingUnderscore: 'require' },`

**Comment by bradzacher** [comment]
> I was getting an error

Again.. you didn't tell me the error.
I cannot repro this against master using the exact config you provided in your OP.
It worked fine and validated fine.

**Comment by pmiatkowski** [comment]
I've updated 'Actual' and 'Expected' results.

**Comment by bradzacher** [comment]
Okay thank you - now with a complete issue, I can understand your problem.
Next time **please follow the issue template**. It's there for a reason so we don't need to waste each other's time with back and forth gathering more information.

---

So your issue is that you provided `[private, protected]`, and expected that that would match *both* private **and** protected members.
This however is not how it works, intentionally so - [as documented in the rule's documentation](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#selector-options):

> The name must match all of the modifiers.

This is done so you can explicitly target specific sets of modifiers - like `[private, static, readonly]` will let you target just `private static readonly` members.
So technically the config `[private, protected]` actually means "match members that are `private protected`, which is nonsensical - so they do not match.

The config is not designed to be DRY - intentionally - it's supposed to be clear, explicit and highly configurable. ESLint configs are just JS - so you're free to add utilities to your config to make it DRY within your codebase.

**Comment by pmiatkowski** [comment]
My apologies for not being clear at first. 
Now I understand this. So it actually works as intended, but my understanding was wrong in this case.
Let's close this one.
